src/openms/include/OpenMS/FORMAT/MSstatsFile.h

```diff
@@ -63,6 +63,21 @@ namespace OpenMS
         ///Destructor
         ~MSstatsFile();
 
+        typedef OpenMS::Peak2D::IntensityType Intensity;
+        typedef OpenMS::Peak2D::CoordinateType Coordinate;
+
+        struct AggregatedConsensusInfo
+        {
+          std::vector< std::vector< String > > consensus_feature_filenames;           // Filenames of ConsensusFeature
+          std::vector< std::vector< Intensity > > consensus_feature_intensites;       // Intensites of ConsensusFeature
+          std::vector< std::vector< Coordinate > > consensus_feature_retention_times; // Retention times of ConsensusFeature
+          std::vector< std::vector< unsigned > > consensus_feature_labels;          // Labels of ConsensusFeature
+        };
+
+        MSstatsFile::AggregatedConsensusInfo aggregateInfo(ConsensusMap const &consensus_map,
+                                                           std::vector<BaseFeature> &features,
```

Comment:
I find it a bit confusing that there are two "results" of this function.
The column vectors (=AggregatedConsensusInfo) and &features

Reply:
could this function and struct be private?

Reply:
The function could and struct could be private. 

And you are right - there should not be two "results". I was not paying enough attention to that while extracting the method. I actually wanted to store all relevant information in AggregatedConsensusInfo.